src/main/kotlin/sliv/tool/scene/view/LandmarkView.kt

```diff
@@ -1,27 +1,153 @@
+        stateSynchronizationManager.landmarkUnselected += landmarkUnselectedEventHandler
+        stateSynchronizationManager.landmarkMouseEntered += landmarkMouseEnteredEventHandler
+        stateSynchronizationManager.landmarkMouseExited += landmarkMouseExitedEventHandler
+    }
+
+    fun dispose() {
+        stateSynchronizationManager.landmarkSelected -= landmarkSelectedEventHandler
+        stateSynchronizationManager.landmarkUnselected -= landmarkUnselectedEventHandler
+        stateSynchronizationManager.landmarkMouseEntered -= landmarkMouseEnteredEventHandler
+        stateSynchronizationManager.landmarkMouseExited -= landmarkMouseExitedEventHandler
+    }
+
+    // Set up common shape properties
+    // Can not be called during LandmarkView initialization because shape is created by inheritors
+    protected fun setUpShape(shape: Shape) {
+        if (stateSynchronizationManager.selectedLandmarksUids.contains(landmark.uid)) {
+            state = LandmarkState.Selected
+        }
+
+        shape.addEventHandler(MouseEvent.MOUSE_CLICKED) {
+            when (state) {
+                LandmarkState.Ordinary -> {
+                    select()
+                    stateSynchronizationManager.landmarkSelected.invoke(
+                        LandmarkEventArgs(landmark.uid, landmark.layer, frameTimestamp)
+                    )
+                }
+                LandmarkState.Selected -> {
+                    unselect()
+                    stateSynchronizationManager.landmarkUnselected.invoke(
+                        LandmarkEventArgs(landmark.uid, landmark.layer, frameTimestamp)
+                    )
+                }
+            }
+        }
+
+        shape.addEventHandler(MouseEvent.MOUSE_ENTERED) {
+            if (state == LandmarkState.Selected) {
+                return@addEventHandler
+            }
+
+            handleMouseEntered()
+            stateSynchronizationManager.landmarkMouseEntered.invoke(
+                LandmarkEventArgs(landmark.uid, landmark.layer, frameTimestamp)
+            )
+        }
+
+        shape.addEventHandler(MouseEvent.MOUSE_EXITED) {
+            if (state == LandmarkState.Selected) {
+                return@addEventHandler
+            }
+
+            handleMouseExited()
+            stateSynchronizationManager.landmarkMouseExited.invoke(
+                LandmarkEventArgs(landmark.uid, landmark.layer, frameTimestamp)
+            )
+        }
+    }
+
+    private fun select() {
+        state = LandmarkState.Selected
+        highlightShape()
+    }
+
+    private fun unselect() {
+        state = LandmarkState.Ordinary
+        unhighlightShape()
+    }
+
+    private fun handleMouseEntered() {
+        if (state == LandmarkState.Selected) {
+            return
+        }
+
+        highlightShape()
+    }
+
+    private fun handleMouseExited() {
+        if (state == LandmarkState.Selected) {
+            return
+        }
+
+        unhighlightShape()
```

Comment:
it will highlight with the same color as selected ones? it will be a problem I think

Reply:
I moved colorizing logic from `LineLayer` to `Layer` and generate unique color in the layer class. Colors are unique, but looks ugly